internal/store/driver.go

```diff
@@ -642,8 +852,416 @@ func (d *Driver) calculateHTTPSEdges() map[string]ingressv1alpha1.HTTPSEdge {
+		}
+
+		if len(match.QueryParams) > 0 {
+			d.log.Error(fmt.Errorf("Unsupported match type"), "Unsupported match type", "HTTPQueryParamMatch", match.QueryParams)
+		}
+	}
+
+	responseHeaders := make(map[string]string)
+	for _, filter := range rule.Filters {
+		switch filter.Type {
+		case gatewayv1.HTTPRouteFilterRequestRedirect:
+			// NOTE: request redirect is a special case, and is subject to change
+			err := d.handleRequestRedirectFilter(filter.RequestRedirect, pathPrefixMatches, &inboundActions, responseHeaders)
+			if err != nil {
+				return nil, err
+			}
+		case gatewayv1.HTTPRouteFilterRequestHeaderModifier:
+			err := d.handleHTTPHeaderFilter(filter.RequestHeaderModifier, &inboundActions, nil)
+			if err != nil {
+				return nil, err
+			}
+		case gatewayv1.HTTPRouteFilterResponseHeaderModifier:
+			err := d.handleHTTPHeaderFilter(filter.ResponseHeaderModifier, &outboundActions, responseHeaders)
+			if err != nil {
+				return nil, err
+			}
+		case gatewayv1.HTTPRouteFilterURLRewrite:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unsupported filter HTTPRouteFilterType %v found", filter.Type))
+		case gatewayv1.HTTPRouteFilterRequestMirror:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unsupported filter HTTPRouteFilterType %v found", filter.Type))
+		case gatewayv1.HTTPRouteFilterExtensionRef:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unsupported filter HTTPRouteFilterType %v found", filter.Type))
+		default:
+			return nil, errors.NewErrorNotFound(fmt.Sprintf("Unknown filter HTTPRouteFilterType %v found", filter.Type))
+		}
+	}
+
+	var policy *ingressv1alpha1.EndpointPolicy
+	enabled := true
+
+	if len(expressions) > 1 {
+		expressions = []string{strings.Join(expressions[:], " || ")}
+	}
+
+	if inboundActions != nil {
+		policy = &ingressv1alpha1.EndpointPolicy{
+			Enabled: &enabled,
+			// NOTE: Mapping each HTTPRouteRule to one Inbound endpoint rule
+			Inbound: []ingressv1alpha1.EndpointRule{
+				{
+					Expressions: expressions,
+					Actions:     *inboundActions,
+					Name:        "Inbound HTTPRouteRule",
+				},
+			},
+		}
+	}
+	if outboundActions != nil {
+		if policy == nil {
+			policy = &ingressv1alpha1.EndpointPolicy{
+				Enabled: &enabled,
+			}
+		}
+
+		policy.Outbound = []ingressv1alpha1.EndpointRule{
+			{
+				Expressions: expressions,
+				Actions:     *outboundActions,
+				Name:        "Outbound HTTPRouteRule",
+			},
+		}
+	}
+
+	return policy, nil
+}
+
+type RemoveHeadersConfig struct {
+	Headers []string `json:"headers"`
+}
+
+type AddHeadersConfig struct {
+	Headers map[string]string `json:"headers"`
+}
+
+func (d *Driver) handleHTTPHeaderFilter(filter *gatewayv1.HTTPHeaderFilter, actions **[]ingressv1alpha1.EndpointAction, requestRedirectHeaders map[string]string) error {
+	if *actions == nil {
+		*actions = &[]ingressv1alpha1.EndpointAction{}
+	}
+	if filter != nil {
```

Comment:
reversing this will probably benefit reading 🤷

Reply:
unrolled here: https://github.com/ngrok/kubernetes-ingress-controller/pull/343/commits/e99dce670699fdfc565a6eee0faf57c077c94b35